react-common/components/controls/Input.tsx

```diff
@@ -59,11 +60,13 @@ export const Input = (props: InputProps) => {
         onOptionSelected,
         handleInputRef,
         preserveValueOnBlur,
-        options
+        options,
+        filter
     } = props;
 
-    const [value, setValue] = React.useState(undefined);
+    const [value, setValue] = React.useState(initialValue || "");
     const [expanded, setExpanded] = React.useState(false);
+    const [pattern, setPattern] = React.useState(filter ? new RegExp(filter) : undefined);
```

Comment:
Looks good, except I would avoid using the name `pattern` as the input element has an attribute by this name and we may want to expose it at some point. You could call it `filter` if you didn't destructure that value from props. Something like:
```suggestion
    const [filter] = React.useState(opts.filter ? new RegExp(opts.filter) : undefined);
```